src/Microsoft.Extensions.Logging.Testing/AssemblyTestLog.cs

```diff
@@ -234,6 +253,26 @@ private static SerilogLoggerProvider ConfigureFileLogging(string fileName)
             return new SerilogLoggerProvider(serilogger, dispose: true);
         }
 
+        private static string RemoveIllegalFileChars(string s)
+        {
+            var sb = new StringBuilder();
+
+            foreach (var c in s)
+            {
+                if (InvalidFileChars.Contains(c))
+                {
+                    var escapedChar = Convert.ToByte(c).ToString("x");
```

Comment:
This expands string size a lot, maybe just do '_' ?

Reply:
I did do that at one point but it made test names highly unreadable. Also converting to the same character for all invalid chars creates lots of collisions. With the test name shortening logic in place, I don't think this should be an issue.

Reply:
Hmm right now I expand illegal characters like `/` => `-x2f`, maybe I can reduce the verbosity and just leave it as `2f` which is a bit harder to parse out but still retains most of the information.

Reply:
> I did do that at one point but it made test names highly unreadable

Hex is not very readable too.

> With the test name shortening logic in place, I don't think this should be an issue. 

Replacing a bunch of '_' with one would be a nice addition to shortening logic.

Reply:
Poor choice of words, I meant file name collisions instead of readability. Even though we have collision resolution, I would prefer retaining as much information as possible regarding the arguments used by default.

Reply:
Do you really prefer `Testing_https:-x2f-x2flocalhost` over `Testing_https:_localhost` ? To me, the second one is much easier to figure out.

Reply:
For that case maybe, but this will create many collisions for kestrel tests, for example: https://github.com/aspnet/KestrelHttpServer/blob/dev/test/shared/HttpParsingData.cs#L192. Also I don't like coalescing multiple `_` since that makes tests like https://github.com/aspnet/KestrelHttpServer/blob/dev/test/shared/HttpParsingData.cs#L254-L255 indistiguishable.

Reply:
@halter73 what would you prefer?